src/Tuura/Concept/Concat/Abstract.hs

```diff
@@ -1,51 +1,57 @@
 module Tuura.Concept.Concat.Abstract (
     Interface (..),
+    InitialValue (..),
     Concept (..),
     initialConcept, arcConcept,
     interfaceConcept,
-    (.&&.), (.||.),
     ) where
 
-import Control.Applicative
+import Data.Monoid
 
 -- Abstract concepts
 -- * s is the type of states
 -- * e is the type of events
 
 data Interface = Unused | Input | Internal | Output deriving (Ord, Eq, Show)
 
+data InitialValue = Undefined | Defined { getDefined :: Bool } | Inconsistent deriving (Eq, Show)
+
+instance Monoid InitialValue where
+    mempty = Undefined
+
+    mappend Inconsistent _ = Inconsistent
+    mappend _ Inconsistent = Inconsistent
+    mappend Undefined x = x
+    mappend x Undefined = x
+    mappend (Defined x) (Defined y) = if x == y then Defined x else Inconsistent
+
+
 data Concept s e a = Concept
                    {
-                       initial :: s -> Bool,
+                       initial :: a -> InitialValue,
                        arcs    :: [(e, e)],
                        interface :: a -> Interface
                    }
 
 instance Monoid (Concept s e a) where
     mempty = Concept
              {
-                 initial = const True,
+                 initial = const mempty,
                  arcs    = [],
                  interface = const Unused
```

Comment:
For symmetry, I suggest to add a `Monoid` instance for `Interface`, so we can say here `const mempty` and use `<>` instead of `max` lateron. This makes this more iniform.